google/cloud/aiplatform/metadata/resource.py

```diff
@@ -38,7 +38,8 @@ class _Resource(base.AiPlatformResourceNounWithFutureManager, abc.ABC):
 
     def __init__(
         self,
-        resource_name: str,
+        resource_name: str = None,
+        resource: proto.Message = None,
```

Comment:
This type should probably be scoped down to a Union of the Artifact, Context, Execution protos. Also it's optional here.

Reply:
Done